build/android/escargot/src/main/cpp/JNIErrorObject.cpp

```diff
@@ -53,11 +53,46 @@ Java_com_samsung_lwe_escargot_JavaScriptErrorObject_create(JNIEnv* env, jclass c
     StringRef* jsMessage = createJSStringFromJava(env, message);
 
     auto contextRef = getPersistentPointerFromJava<ContextRef>(env, env->GetObjectClass(context), context);
-    auto evaluatorResult = Evaluator::execute(contextRef->get(), [](ExecutionStateRef* state,
+    auto evaluatorResult = ScriptEvaluator::execute(contextRef->get(), [](ExecutionStateRef* state,
                                                                     ErrorObjectRef::Code code, StringRef* jsMessage) -> ValueRef* {
-        return ErrorObjectRef::create(state, code, jsMessage);
+        ErrorObjectRef* errorObject = ErrorObjectRef::create(state, code, jsMessage);
+        std::string stacktrace = ExecutionStateRefTracker::computeStacktrace();
+        errorObject->defineDataProperty(state, StringRef::createFromASCII("stack"),
+                                        ObjectRef::DataPropertyDescriptor(
+                                                StringRef::createFromUTF8(stacktrace.data(), stacktrace.size()),
+                                                ObjectRef::PresentAttribute::AllPresent));
+        return errorObject;
     }, code, jsMessage);
 
     assert(evaluatorResult.isSuccessful());
-    return createJavaObjectFromValue(env, evaluatorResult.result->asErrorObject());
+
+    ErrorObjectRef* object = evaluatorResult.result->asErrorObject();
+    ScriptEvaluator::execute(contextRef->get(), [](ExecutionStateRef* state,
+                                                                          ErrorObjectRef* object) -> ValueRef* {
+        std::string stacktrace = ExecutionStateRefTracker::computeStacktrace();
+        object->defineDataProperty(state, StringRef::createFromASCII("stack"),
+                                        ObjectRef::DataPropertyDescriptor(
+                                                StringRef::createFromUTF8(stacktrace.data(), stacktrace.size()),
+                                                ObjectRef::PresentAttribute::AllPresent));
```

Comment:
It seems like this stack calculation of `ErrorObject` is done twice when each ErrorObject is created.
Is this intended?

Reply:
Just creating ErrorObject not compute stacktrace.
since Java API doesn't expose Evaluator & ExecuteState interface on native.
each API call have separate ExecuteState chain.
that's why we would compute Stacktrace on JNI layer

Reply:
I see. but it still seems that creating stack trace info for ErrorObject is done **twice** when this ErrorObject creation API called, first when ErrorObject created and then just before returning the creating ErrorObject. They exactly do the same thing, creating stack trace info and defining it as `stack` property. I think that the second stack creation can be removed.

Reply:
Opps